src/compiler/corePublic.ts

```diff
@@ -117,22 +117,30 @@ namespace ts {
         declare const Map: MapConstructor | undefined;
         declare const Set: SetConstructor | undefined;
 
+        const globals = typeof globalThis !== "undefined" ? globalThis
+                  : typeof global !== "undefined" ? global
+                  // @ts-ignore
+                  : typeof window !== "undefined" ? window

```

Comment:
Use `self` instead of `window`. `self` works on both web pages and web workers, while `window` only works on web pages.
```suggestion
                  : typeof self !== "undefined" ? self
```

Reply:
you could also do `declare const self: any` so you don't need to have the `// @ts-ignore`